cmd/aries-agentd/startcmd/start.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/spf13/cobra"
+	"net/http"
+	"strings"
+)
+
+const AgentHostFlagName = "api-host"
+const AgentHostFlagShorthand = "a"
+const AgentHostFlagUsage = "Host Name:Port"
+
+const AgentInboundHostFlagName = "inbound-host"
+const AgentInboundHostFlagShorthand = "i"
+const AgentInboundHostFlagUsage = "Inbound Host Name:Port"
+
+const AgentDBPathFlagName = "db-path"
+const AgentDBPathFlagShorthand = "d"
+const AgentDBPathFlagUsage = "Path to database"
+
+const MissingHostErrorMessage = "Unable to start aries agentd, host not provided"
+const MissingInboundHostErrorMessage = "Unable to start aries agentd, HTTP Inbound transport host not provided"
+
+var logger = log.New("aries-framework/agentd")
+
+type server interface {
+	ListenAndServe(host string, router *mux.Router) error
+}
+
+type RealServer struct {
+	server http.Server
+}
+
+func (s *RealServer) ListenAndServe(host string, router *mux.Router) error {
+	return http.ListenAndServe(host, router)
+}
+
+func Cmd(server server) *cobra.Command {
+	startCmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start an agent",
+		Long:  `Start an Aries agent controller`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			host, err := cmd.Flags().GetString(AgentHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			inboundHost, err := cmd.Flags().GetString(AgentInboundHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			dbPath, err := cmd.Flags().GetString(AgentDBPathFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			err = startAgent(server, host, inboundHost, dbPath)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+		},
+	}
+	startCmd.Flags().StringP(AgentHostFlagName, AgentHostFlagShorthand, "", AgentHostFlagUsage)
+	startCmd.MarkFlagRequired(AgentHostFlagName)
+	startCmd.Flags().StringP(AgentInboundHostFlagName, AgentInboundHostFlagShorthand, "", AgentInboundHostFlagUsage)
+	startCmd.MarkFlagRequired(AgentInboundHostFlagName)
+	startCmd.Flags().StringP(AgentDBPathFlagName, AgentDBPathFlagShorthand, "", AgentDBPathFlagUsage)
+	startCmd.MarkFlagRequired(AgentDBPathFlagName)
+
+	return startCmd
+}
+
+func startAgent(server server, host string, inboundHost string, dbPath string) error {
+	if host == "" {
+		return errors.New(strings.ToLower(MissingHostErrorMessage))
+	}
+
+	var opts []aries.Option
+	if inboundHost == "" {
+		return errors.New(strings.ToLower(MissingInboundHostErrorMessage))
+	}
+	opts = append(opts, defaults.WithInboundHTTPAddr(inboundHost))
+
+	if dbPath != "" {
+		opts = append(opts, defaults.WithStorePath(dbPath))
+	}
+
+	framework, err := aries.New(opts...)
+	if err != nil {
+		return fmt.Errorf("failed to start aries agentd on port [%s], failed to initialize framework :  %w", host, err)
+	}
+
+	ctx, err := framework.Context()
+	if err != nil {
+		return fmt.Errorf("failed to start aries agentd on port [%s], failed to get aries context : %w", host, err)
```

Comment:
Unsure how to unit test line 110. Aries framework is created through controlled inputs. Note also that the existing test cases didn't test this (was originally in main.go)